doc/lsst.pex.config/registry-intro.rst

```diff
@@ -0,0 +1,29 @@
+.. py:currentmodule:: lsst.pex.config
+
+###################################################
+Registry and RegistryField for configuring subtasks
+###################################################
```

Comment:
Using `RegistryField` for subtasks is a bit of an edge case (usually we'd use `ConfigurableField`), so while this is an important example, it's a bit jarring to have it without a `ConfigurableField` example.

We should probably pull out and expand the bit about the `Configurable` concept from the `RegistryField` docs and make it a topic page that includes examples of both `ConfigurableField` and `RegistryField`.  That's obviously not a job for this ticket, but we should make a ticket for it.